packages/atlas/src/views/global/YppLandingView/YppAuthorizationModal/YppAuthorizationModal.hooks.tsx

```diff
@@ -0,0 +1,150 @@
+import { useCallback, useEffect } from 'react'
+import { useSearchParams } from 'react-router-dom'
+
+import { GOOGLE_CONSOLE_CLIENT_ID, GOOGLE_OAUTH_ENDPOINT } from '@/config/env'
+import { useConfirmationModal } from '@/providers/confirmationModal'
+import { useSnackbar } from '@/providers/snackbars'
+import { useYppStore } from '@/providers/ypp/ypp.store'
+import { createId } from '@/utils/createId'
+import { SentryLogger } from '@/utils/logs'
+
+const GOOGLE_AUTH_PARAMS = {
+  client_id: GOOGLE_CONSOLE_CLIENT_ID,
+  response_type: 'code',
+  scope: 'https://www.googleapis.com/auth/youtube.force-ssl',
+  include_granted_scopes: 'true',
+  prompt: 'consent',
+}
+
+export const useYppGoogleAuth = ({
+  closeModal,
+  goToLoadingStep,
+  selectedChannelId,
+  setSelectedChannelId,
+  channelsLoaded,
+}: {
+  closeModal: () => void
+  goToLoadingStep: () => void
+  selectedChannelId: string | null
+  setSelectedChannelId: (channelId: string | null) => void
+  channelsLoaded: boolean
+}) => {
+  const oldAuthState = useYppStore((state) => state.authState)
+  const setAuthState = useYppStore((state) => state.actions.setAuthState)
+
+  const [openConfirmationModal, closeConfirmationModal] = useConfirmationModal()
+  const { displaySnackbar } = useSnackbar()
+
+  const [searchParams, setSearchParams] = useSearchParams()
+  const resetSearchParams = useCallback(() => setSearchParams(new URLSearchParams()), [setSearchParams])
+
+  const handleAuthorizeClick = useCallback(() => {
+    const authUrl = new URL(GOOGLE_OAUTH_ENDPOINT)
+    const randomCode = createId()
+
+    const stateParams = new URLSearchParams({
+      code: randomCode,
+      channelId: selectedChannelId || '',
+    })
+    const authState = stateParams.toString()
+    setAuthState(authState)
+
+    const authParams = {
+      ...GOOGLE_AUTH_PARAMS,
+      redirect_uri: window.location.href,
+      state: authState,
+    }
+    Object.entries(authParams).forEach(([key, value]) => authUrl.searchParams.set(key, value))
+
+    window.location.assign(authUrl)
+  }, [selectedChannelId, setAuthState])
+
+  const handleGoogleAuthError = useCallback(
+    (error: string) => {
+      // Google auth failed, show error modal
+      SentryLogger.error('Google Auth failed', 'YppAuthorizationModal', error)
+
+      // TODO: use proper copy
+      openConfirmationModal({
+        title: 'Authorization failed',
+        description: <>Google Auth failed</>,
+        type: 'destructive',
+        primaryButton: {
+          text: 'Try again',
+          onClick: () => {
```

Comment:
Shouldn't we start authorization flow from the beginning? In fact it does the same as cancel button